cubi_tk/sodar/update_samplehseet.py

```diff
@@ -0,0 +1,628 @@
+        if not self.args.ped and not self.args.sample_data:
+            raise ValueError(
+                "No sample data provided. Please provide either a ped file (`--ped`) or sample data (`-s`)."
+            )
+
+        # Base field name mapping (also used by ped)
+        sample_field_mapping = sheet_default_config[self.args.defaults]["field_column_mapping"]
+
+        # Sample data, if given, need to matche the defined fields (from default or sample_fields)
+        if self.args.sample_data:
+            if self.args.sample_fields:
+                sample_field_mapping.update(
+                    OrderedDict(
+                        [
+                            (field.split("=")[0], field.split("=")[1 if "=" in field else 0])
+                            for field in self.args.sample_fields
+                        ]
+                    )
+                )
+                n_fields = len(self.args.sample_fields)
+            else:
+                n_fields = len(sheet_default_config[self.args.defaults]["sample_fields"])
+
+            incorrect_n = [sample for sample in self.args.sample_data if len(sample) != n_fields]
+            if incorrect_n:
+                msg = (
+                    f"The number of entries for some samples does not match the number of fields defined ({n_fields}):\n"
+                    "\n".join(
+                        [
+                            f"{len(sample)} instead of {n_fields} values: {', '.join(map(str, sample))}"
+                            for sample in incorrect_n
+                        ]
+                    )
+                )
+                raise ValueError(msg)
+
+            unknown_fields = [
+                tup for tup in sample_field_mapping.items() if tup[1] not in isa_names.keys()
+            ]
+            if unknown_fields:
+                msg = (
+                    "Some columns for sample field mapping are not defined in the ISA samplesheet: "
+                    + ", ".join([f"{col} (from {field})" for field, col in unknown_fields])
+                )
+                raise NameError(msg)
+
+        return sample_field_mapping
+
+    def gather_ISA_column_names(
+        self, study: pd.DataFrame, assay: pd.DataFrame
+    ) -> dict[str, list[tuple[str, str]]]:
+        isa_regex = re.compile(r"(Characteristics|Parameter Value|Comment)\[(.*?)]")
+        study_cols = study.columns.tolist()
+        assay_cols = assay.columns.tolist()
+
+        isa_short_names = list(
+            map(lambda x: orig_col_name(isa_regex.sub(r"\2", x)), study_cols + assay_cols)
+        )
+        isa_long_names = list(map(orig_col_name, study_cols + assay_cols))
+        isa_names_unique = study_cols + assay_cols
+        isa_table = ["study"] * len(study_cols) + ["assay"] * len(assay_cols)
+
+        out = defaultdict(list)
+        for short, long, uniq, table in zip(
+            isa_short_names, isa_long_names, isa_names_unique, isa_table
+        ):
```

Comment:
Add `strict=True` to the arguments of `zip` to ensure they're all of the same length